pkgs/by-name/ud/udev-hid-bpf/package.nix

```diff
@@ -0,0 +1,91 @@
+{
+  bpftools,
+  clang,
+  elfutils,
+  fetchFromGitLab,
+  lib,
+  libbpf,
+  meson,
+  ninja,
+  pkg-config,
+  python3,
+  python3Packages,
+  rustPlatform,
+  systemd,
+  zlib,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "udev-hid-bpf";
+  version = "2.2.0-20251121";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = "libevdev";
+    repo = pname;
+    rev = "${version}";
+    hash = "sha256-zZy0qjwirKYbNKFPt3SoM9993h6jOzfmg9jet/7vJ6U=";
+  };
+
+  cargoHash = "sha256-cf5FLPT581Dy4msM1hdvflN1OHij+sVdZwLHyRRYWw8=";
+
+  nativeBuildInputs = [
+    bpftools
+    clang
+    meson
+    ninja
+    pkg-config
+    python3
+  ];
+
+  buildInputs = [
+    elfutils
+    libbpf
+    systemd.dev
+    zlib
+  ];
+
+  nativeCheckInputs = with python3Packages; [
+    click
+    gitpython
+    pytest
+  ];
+
+  dontUseNinjaBuild = true;
+  dontUseNinjaCheck = true;
+  dontUseNinjaInstall = true;
+
+  # mesonFlags = [
+  #   "-Dprefix=$out"
+  #   "-Dudev-dir=$out/etc"
+  # ];
+
+  patchPhase = ''
+    sed -i '1s|.*|#!${python3}/bin/python3|' tools/generate-hwdb.py
+    sed -i '266s|libbpf\.so|${libbpf}/lib/libbpf.so|' test/btf.py
```

Comment:
Something like:

```
    patchShebangs tools/generate-hwdb.py
    substituteInPlace test/btf.py \
      --replace-fail "'/usr/include/@0@'.format(target_triplet)" "'${lib.getInclude linuxHeaders}/include'"
```

Using `sed` directly is fragile and doesn't catch problems when the files are updated